awswrangler/s3/_read_parquet.py

```diff
@@ -236,457 +216,269 @@ def _read_parquet_metadata(
-            ignore_metadata=True,
-            strings_to_categorical=False,
-            safe=safe,
-            categories=categories,
-            types_mapper=_data_types.pyarrow2pandas_extension if map_types else None,
-        ),
-        dataset=dataset,
-        path=path,
-        path_root=path_root,
-    )
-    df = _utils.ensure_df_is_mutable(df=df)
-    if metadata:
-        _logger.debug("metadata: %s", metadata)
-        df = _apply_timezone(df=df, metadata=metadata)
-        df = _apply_index(df=df, metadata=metadata)
-    return df
-
-
-def _pyarrow_chunk_generator(
-    pq_file: pyarrow.parquet.ParquetFile,
-    chunked: Union[bool, int],
     columns: Optional[List[str]],
-    use_threads_flag: bool,
-) -> Iterator[pa.RecordBatch]:
-    if chunked is True:
-        batch_size = 65_536
-    elif isinstance(chunked, int) and chunked > 0:
-        batch_size = chunked
-    else:
-        raise exceptions.InvalidArgument(f"chunked: {chunked}")
-
-    chunks = pq_file.iter_batches(
-        batch_size=batch_size, columns=columns, use_threads=use_threads_flag, use_pandas_metadata=False
-    )
-
-    for chunk in chunks:
-        yield chunk
-
-
-def _row_group_chunk_generator(
-    pq_file: pyarrow.parquet.ParquetFile,
-    columns: Optional[List[str]],
-    use_threads_flag: bool,
-    num_row_groups: int,
-) -> Iterator[pa.Table]:
-    for i in range(num_row_groups):
-        _logger.debug("Reading Row Group %s...", i)
-        yield pq_file.read_row_group(i=i, columns=columns, use_threads=use_threads_flag, use_pandas_metadata=False)
+    coerce_int96_timestamp_unit: Optional[str],
+    s3_additional_kwargs: Optional[Dict[str, str]],
+    use_threads: Union[bool, int],
+    version_id: Optional[str] = None,
+) -> pa.Table:
+    s3_block_size: int = FULL_READ_S3_BLOCK_SIZE if columns else -1  # One shot for a full read or see constant
+    with open_s3_object(
+        path=path,
+        mode="rb",
+        version_id=version_id,
+        use_threads=use_threads,
+        s3_block_size=s3_block_size,
+        s3_additional_kwargs=s3_additional_kwargs,
+        boto3_session=boto3_session,
+    ) as f:
+        pq_file: Optional[pyarrow.parquet.ParquetFile] = _pyarrow_parquet_file_wrapper(
+            source=f,
+            coerce_int96_timestamp_unit=coerce_int96_timestamp_unit,
+        )
+        if pq_file is None:
+            raise exceptions.InvalidFile(f"Invalid Parquet file: {path}")
+        return _add_table_partitions(
+            table=pq_file.read(columns=columns, use_threads=False, use_pandas_metadata=False),
+            path=path,
+            path_root=path_root,
+        )
 
 
-def _read_parquet_chunked(  # pylint: disable=too-many-branches
-    paths: List[str],
-    chunked: Union[bool, int],
-    validate_schema: bool,
-    ignore_index: Optional[bool],
-    columns: Optional[List[str]],
-    categories: Optional[List[str]],
-    safe: bool,
-    map_types: bool,
+def _read_parquet_chunked(
```

Comment:
Is my understanding correct here that the refactoring in this method is:
1. Schema validation was removed
2. Chunking case for pyarrow < 3 was removed as we're supporting from version 6 now

Has the validation been moved somewhere else?

Reply:
Ok found the validation

Reply:
Yeah, validation is now centralised and done at the very [beginning](https://github.com/awslabs/aws-data-wrangler/blob/feat-3.0/distributed-s3-read-parquet/awswrangler/s3/_read_parquet.py#L544). Before it was done in multiple places and done much later

Reply:
And yes, I think it's time to drop arrow < 3, so I suggest we also drop any logic that handled older versions

Reply:
Do all our major runtimes support pyarrow above 3 now? Like, Glue, Lambda?

Reply:
Lambda and Python Shell for sure, not sure about the rest. I created an [issue](https://github.com/awslabs/aws-data-wrangler/issues/1518) so one of us can check. To be clear though, ray and modin only support pyarrow 6+, so we would need to some gymnastics in the pyproject.toml to support older versions...